[ONNX] Clean up diagnostic rules

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #107654
* __->__ #107653
* #107409
* #107408

Summary:

- Remove experimental rules that were never used.
- Fill in detailed rule descriptions.

BowenBao committed:
[ONNX] Clean up diagnostic rules

Summary:

- Remove experimental rules that were never used.
- Fill in detailed rule descriptions.

[ghstack-poisoned]

BowenBao commented on torch/onnx/_internal/diagnostics/_rules.py:
This is codegen file, please ignore.

BowenBao commented on torch/csrc/onnx/diagnostics/generated/rules.h:
This is codegen file, please ignore.

titaiwangms commented on torch/onnx/_internal/diagnostics/rules.yaml:
```diff
@@ -83,178 +79,98 @@
-
-- id: FXE0001
-  name: fx-tracer-success
+- id: FXE0007
```
Should we re-start from one?

BowenBao replied:
I'm leaving as is, technically it is taken and exposed in 2.0 ..

BowenBao committed:
adjust package name; remove xfail for unexpected pass on "[ONNX] Clean up diagnostic rules"


Summary:

- Remove experimental rules that were never used.
- Fill in detailed rule descriptions.

[ghstack-poisoned]

BowenBao committed:
use commit from onnxscript main in CI on "[ONNX] Clean up diagnostic rules"


Summary:

- Remove experimental rules that were never used.
- Fill in detailed rule descriptions.

[ghstack-poisoned]

BowenBao committed:
adjust xfail removal between commits; enable sarif log saving for unittest when logs enabled on "[ONNX] Clean up diagnostic rules"


Summary:

- Remove experimental rules that were never used.
- Fill in detailed rule descriptions.

[ghstack-poisoned]

BowenBao committed:
remove more xfails on "[ONNX] Clean up diagnostic rules"


Summary:

- Remove experimental rules that were never used.
- Fill in detailed rule descriptions.

[ghstack-poisoned]
